dist/pythonlibs/riotctrl_shell/loramac.py

```diff
@@ -0,0 +1,96 @@
+# directory for more details.
+
+"""
+Shell interactions related to Semtech Loramac package
+
+Defines shell command interactions related to Semtech Loramac package
+"""
+
+import re
+
+from riotctrl.shell import ShellInteraction
+
+# ==== Parsers ====
+
+
+class LoramacHelpParser():
+    def __init__(self):
+        self.eeprom_c = re.compile(r'Usage: loramac \<([\w+\|?]+)\>')
+
+    def has_eeprom(self, cmd_output):
+        for line in cmd_output.splitlines():
+            m = self.eeprom_c.search(line)
+            if m is not None:
+                if 'save' in m.group(1):
+                    return True
+        return False
+
+
+class LoramacUpLinkCounterParser():
+    def __init__(self):
+        self.uplink_c = re.compile(r'Uplink Counter: (\d+)')
+
+    def uplink_count(self, cmd_output):
+        for line in cmd_output.splitlines():
+            m = self.uplink_c.search(line)
+            if m is not None:
+                return m.group(1)
+        raise RuntimeError
+
+# ==== ShellInteractions ====
+
+
+class Loramac(ShellInteraction):
+    @ShellInteraction.check_term
+    def loramac_cmd(self, args=None, timeout=-1, async_=False):
+        cmd = "loramac"
+        if args is not None:
+            cmd += " {args}".format(args=" ".join(str(a) for a in args))
+        return self.cmd(cmd, timeout=timeout, async_=False)
+
+    def loramac_join(self, mode, timeout=-1, async_=False):
+        res = self.loramac_cmd(args=("join", mode),
+                               timeout=timeout, async_=async_)
+        if "succeeded" in res:
+            return res
+        raise RuntimeError(res)
+
+    def loramac_tx(self, payload, cnf=False, port=2, timeout=-1, async_=False):
+        payload = '\"' + payload + '\"'
+        res = self.loramac_cmd(
+            args=("tx", payload, 'cnf' if cnf else "uncnf", port),
+            timeout=timeout, async_=async_
+            )
+        if "success" in res:
+            return res
+        raise RuntimeError(res)
+
+    def loramac_set(self, key, value, timeout=-1, async_=False):
+        res = self.loramac_cmd(args=("set", key, value),
+                               timeout=timeout, async_=async_)
+        if "Usage" in res:
+            raise RuntimeError(res)
+        return res
+
+    def loramac_get(self, key, timeout=-1, async_=False):
+        res = self.loramac_cmd(args=("get", key),
+                               timeout=timeout, async_=async_)
+        if "Usage" in res:
+            raise RuntimeError(res)
+        return res
+
+    @ShellInteraction.check_term
+    def loramac_eeprom_save(self, timeout=-1, async_=False):
+        return self.cmd("loramac save", timeout, async_)
+
+    @ShellInteraction.check_term
+    def loramac_eeprom_erase(self, timeout=-1, async_=False):
+        return self.cmd("loramac erase", timeout, async_)
+
+    @ShellInteraction.check_term
+    def loramac_help(self, timeout=-1, async_=False):
+        return self.cmd("loramac help", timeout, async_)
```

Comment:
Why can't those use `loramac_cmd`? AFAICS this should be possible:

```suggestion
    @ShellInteraction.check_term
    def loramac_eeprom_save(self, timeout=-1, async_=False):
        return self.loramac_cmd(("save",), timeout, async_)

    @ShellInteraction.check_term
    def loramac_eeprom_erase(self, timeout=-1, async_=False):
        return self.cmd(("erase",), timeout, async_)

    @ShellInteraction.check_term
    def loramac_help(self, timeout=-1, async_=False):
        return self.cmd(("help",), timeout, async_)
```

Reply:
Since there are no args, I didn't think it was needed, but if you prefer it styling wise it would definitely work as well.

Reply:
Well not just styling-wise, but also architecturally. Makes more sense to direct the program flow for all subcommands to one method. This way _if_ there is a change needed (simple example: renaming the `loramac` command for some reason or another), we only need to make minimal changes, and not touch "every" method.

Reply:
done